File: system/src/main/java/com/anjunar/technologyspeaks/hibernate/search/annotations/RestSort.java
package com.anjunar.technologyspeaks.hibernate.search.annotations;

import com.anjunar.technologyspeaks.hibernate.search.DefaultSortProvider;
import com.anjunar.technologyspeaks.hibernate.search.SortProvider;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.METHOD})
public @interface RestSort {

    Class<? extends SortProvider> value() default DefaultSortProvider.class;

}

File: system/src/main/kotlin/com/anjunar/technologyspeaks/hibernate/search/annotations/RestSort.kt
package com.anjunar.technologyspeaks.hibernate.search.annotations

import com.anjunar.technologyspeaks.hibernate.search.SortProvider
import kotlin.reflect.KClass

@Target(AnnotationTarget.PROPERTY)
@Retention(AnnotationRetention.RUNTIME)
annotation class RestSort(val value : KClass<out SortProvider<*,*>>)
